### PR TITLE
Fetch default namespace from the backend

### LINKED ISSUE #726: Fetch deafult namespace from the backend instead of using hardcoded "default"
This will be much more convenient for using in the openshift/kubernetes with privileges limited only to particular namespace.

### PR DESCRIPTION
**Pull request by mmelko**
close #726 
fetching the default namespace is done in useEffects KaotoModal.  Also fetching of deployments were done even before so I added checks to fetch deployments only if there is namespace set. 

There is probably a better way how to do this but I'm not sure yet.

### CODE REVIEW

File: src/components/DeploymentsModal.tsx [hunk 1]
```diff
@@ -56,25 +56,29 @@ export const DeploymentsModal = ({ handleCloseModal, isModalOpen }: IDeployments
...
     // fetch deployments
-    fetchDeployments('no-cache', settings.namespace)
-      .then((output) => {
-        setDeployments(output);
-      })
-      .catch((e) => {
-        throw Error(e);
-      });
+    if(settings.namespace !== '') {
+      console.log('fetcging namespace',settings.namespace)
```
Comment (Delawen) [inline src/components/DeploymentsModal.tsx:60]: A bit nitpicky but there is a typo here :)

File: src/components/DeploymentsModal.tsx [hunk 2]
```diff
@@ -56,25 +56,29 @@ export const DeploymentsModal = ({ handleCloseModal, isModalOpen }: IDeployments
...
+            throw Error(e);
+          });
+    }
   }, [settings.namespace]);
 
   // on changes to deployment, re-fetch list of deployments
   useEffect(() => {
     // fetch deployments
+    if(settings.namespace !== '') {
     fetchDeployments('no-cache', settings.namespace)
```
Comment (Delawen) [inline src/components/DeploymentsModal.tsx:75]: More nitpicky: shouldn't this have some indentation?

**Comment by Delawen** [comment]
LGTM

File: src/components/DeploymentsModal.tsx [hunk 3]
```diff
@@ -56,25 +56,30 @@ export const DeploymentsModal = ({ handleCloseModal, isModalOpen }: IDeployments
...
     // fetch deployments
-    fetchDeployments('no-cache', settings.namespace)
-      .then((output) => {
-        setDeployments(output);
-      })
-      .catch((e) => {
-        throw Error(e);
-      });
+    if (settings.namespace !== '') {
+      console.log('fetching namespace', settings.namespace)
```
Comment (kahboom) [inline src/components/DeploymentsModal.tsx:60]: @mmelko - I think you left this here by mistake maybe?

Comment (mmelko) [reply src/components/DeploymentsModal.tsx]: yep, thanks! 
It should be removed now

**Review by kahboom — APPROVED** [review]

**Pull request merged**